src/expressions/operators/compares/Compare.ts

```diff
@@ -32,6 +46,21 @@ class Compare extends Expression {
 			case 'greaterThanOrEqualOp':
 			case 'greaterThanOp':
 				this._compare = 'generalCompare';
+				if (
+					firstType &&
```

Comment:
In what cases are these absent? Could it make sense to default to item()* to prevent these additional checks?

Reply:
Compare is constructed here in `compileAstToExpression.ts` on 435

```
// Binary compare (=, !=, le, is, <<, >>, etc)
function compare(ast: IAST, compilationOptions: CompilationOptions) {
	const firstOperand = astHelper.followPath(ast, ['firstOperand', '*']);
	const secondOperand = astHelper.followPath(ast, ['secondOperand', '*']);
	const firstType = astHelper.getAttribute(firstOperand, 'type');
	const secondType = astHelper.getAttribute(secondOperand, 'type');

	return new Compare(
		ast[0],
		compile(firstOperand, disallowUpdating(compilationOptions)),
		compile(secondOperand, disallowUpdating(compilationOptions)),
		firstType,
		secondType
	);
}```

When the type of the `firstOperand` and `secondOperand` cannot be determined, the `getAttribute` function returns null.